src/wp-includes/class-wp-theme.php

```diff
@@ -1460,6 +1460,30 @@ public function is_allowed( $check = 'both', $blog_id = null ) {
 		return false;
 	}
 
+	/**
+	 * Returns whether this theme is a block-based theme or not.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return bool
+	 */
+	public function is_block_based() {
```

Comment:
How does this differ to `wp_is_block_template_theme()`? Should we maybe delete `wp_is_block_template_theme()` and instead use `wp_get_theme()->is_block_based()`?

Reply:
We should also ideally add a unit test for this new method.

Reply:
> How does this differ to `wp_is_block_template_theme()`?

`wp_is_block_template_theme` allows checking the current theme only.
`WP_Theme::is_block_based` works for any instance of the `WP_Theme` class.

> Should we maybe delete wp_is_block_template_theme() and instead use wp_get_theme()->is_block_based()?

I still see value in this function. It's a handy alternative to calling `wp_get_theme()->is_block_based()` directly.
Could we possibly keep it? I agree that we don't have to duplicate the implementation, so now it uses `wp_get_theme()->is_block_based()` internally.
Would you please let me know what you think of this?

> We should also ideally add a unit test for this new method.

I 100% agree, and I encourage writing unit tests. Yesterday I had to leave early, and I didn't have time for it.
Fixed in 00cda176deaedf0642f0a4be53dbcd9d8a29df7f